python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/router-bgp.j2

```diff
@@ -1473,6 +1473,15 @@ router bgp {{ router_bgp.as }}
 {%         endif %}
 {%         if vrf.evpn_multicast is arista.avd.defined(true) %}
       evpn multicast
+{%             if vrf.evpn_multicast_gateway_dr_election.algorithm is arista.avd.defined %}
+{%                 if vrf.evpn_multicast_gateway_dr_election.algorithm == "preference" and vrf.evpn_multicast_gateway_dr_election.preference_value is arista.avd.defined %}
+{%                     set election_cli = "gateway dr election algorithm preference " ~ vrf.evpn_multicast_gateway_dr_election.preference_value %}
+{%                 else %}
+{%                     set election_cli = "gateway dr election algorithm " ~ vrf.evpn_multicast_gateway_dr_election.algorithm %}
+{%                 endif %}
+         {{ election_cli }}
+         !
+{%             endif %}
```

Comment:
The logic here breaks if I set `algorithm: "preference"` but do not set a `preference_value`. You probably split the `and` into another nested `if`

Reply:
Added inner if for preference_value.